java/vector/src/main/java/org/apache/arrow/vector/BitVectorHelper.java

```diff
@@ -319,4 +321,79 @@ static void setBitMaskedByte(ArrowBuf data, int byteIndex, byte bitMask) {
     currentByte |= bitMask;
     data.setByte(byteIndex, currentByte);
   }
+
+  /**
+   * Concat two validity buffers.
+   * @param input1 the first validity buffer.
+   * @param numBits1 the number of bits in the first validity buffer.
+   * @param input2 the second validity buffer.
+   * @param numBits2 the number of bits in the second validity buffer.
+   * @param output the ouput validity buffer. It can be the same one as the first input.
```

Comment:
this needs to be preallocated?

Reply:
Sure. I have stated this explicitly in the JavaDoc.